### PR TITLE
feat: Procedure to create table and register table to catalog

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
This PR
- Implements a procedure `CreateTableProcedure` to create a table and register the table to catalog.
- Adds a new crate `table-procedure` for table-related procedures.
- Add options to use procedure to create table (disabled by default).

Configuration section example:
```toml
[procedure.store]
type = 'File'
data_dir = '/path/to/procedure/'
```

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
- #1035
- #1043 
- #286

### CODE REVIEW

File: src/table-procedure/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+[package]
+name = "table-procedure"
+version = "0.1.0"
+edition = "2021"
```
Comment (MichaelScofield) [inline src/table-procedure/Cargo.toml:4]:
```suggestion
version.workspace = true
edition.workspace = true
license.workspace = true
```

File: src/table-procedure/src/create.rs [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+                        engine_procedure.clone(),
+                    )
+                    .map(|p| Box::new(p) as _)
+                }),
+            )
+            .unwrap()
+    }
+
+    /// Recover the procedure from json.
+    fn from_json(
```
Comment (MichaelScofield) [inline src/table-procedure/src/create.rs:117]: Is it better to uplift deserialization method to the procedure trait? There's serialization method `dump` in that trait, which should be paired with some deserialization to stay consistent with each other I think.

Comment (evenyag) [reply src/table-procedure/src/create.rs:117]: Different procedures require different components (e.g. catalog manager, table engine) so it's hard to define a general deserialize method.

File: src/table-procedure/src/create.rs [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+                    logging::info!(
+                        "On engine create table {}, done, sub_id: {}",
+                        self.data.request.table_name,
+                        sub_id
+                    );
+                    // The sub procedure is done, we can execute next step.
+                    self.data.state = CreateTableState::RegisterCatalog;
+                    Ok(Status::executing(true))
+                }
+                ProcedureState::Failed => {
```
[the remark this answers is not in the document]
Comment (evenyag) [reply src/table-procedure/src/create.rs:190]: Change to returning Err if the subprocedure is failed

**Review by MichaelScofield — APPROVED** [review]

**Review by v0y4g3r — APPROVED** [review]
L6TM

**Pull request merged**